tests/Core/TimerFactoryUnitTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Chronos.Timer.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+
+namespace Chronos.Timer.Tests.Core
+{
+    [TestClass]
+    public class TimerFactoryUnitTests
+    {
+        private readonly TimerFactory _target;
+
+        public TimerFactoryUnitTests()
+        {
+            _target = new TimerFactory();
+        }
+
+        [TestMethod]
+        public void CreateTimer_TimeTrackingIsNull_CreatesValidNewTimer()
+        {
+            TestTimer timer = _target.CreateTimer<TestTimer>() as TestTimer;
+
+            AssertValidCreation(timer);
+            Assert.IsNull(timer.TimeTrackingStrategy);
+            Assert.AreEqual(0, timer.Tasks.Count);
+        }
+
+        [TestMethod]
+        public void CreateTimer_CreateTwoTimer_CreatesDifferentTimers()
+        {
+            ITimer timer1 = _target.CreateTimer<TestTimer>();
+            ITimer timer2 = _target.CreateTimer<TestTimer>();
+
+            Assert.AreNotEqual(timer1, timer2);
+        }
+
+        [TestMethod]
+        public void CreateTimer_SystemTimeTracking_CreatesValidNewTimer()
+        {
+            TestTimer timer = _target.CreateTimer<TestTimer>(new SystemTimeTracker()) as TestTimer;
+
+            AssertValidCreation(timer);
+            Assert.IsInstanceOfType(timer.TimeTrackingStrategy, typeof(SystemTimeTracker));
+        }
+
+        [TestMethod]
+        public void CreateTimer_NullTimerTask_CreatesValidNewTimer()
+        {
+            TestTimer timer = _target.CreateTimer<TestTimer>(new SystemTimeTracker()) as TestTimer;
+
+            AssertValidCreation(timer);
+            Assert.AreEqual(0, timer.Tasks.Count);
+        }
+
+        [TestMethod]
+        public void CreateTimer_SingleTimerTask_CreatesValidNewTimer()
+        {
+            TestTimer timer = _target.CreateTimer<TestTimer>(
+                new TimerTask(() => { }, TimeSpan.FromSeconds(1)),
+                new SystemTimeTracker()) as TestTimer;
+
+            AssertValidCreation(timer);
+            Assert.AreEqual(1, timer.Tasks.Count);
+        }
+
+        [TestMethod]
+        public void CreateTimer_MultipleTimerTask_CreatesValidNewTimer()
+        {
+            List<ITimerTask> tasks = new List<ITimerTask>
+            {
+                new TimerTask(() => { }, TimeSpan.FromSeconds(1)),
+                new TimerTask(() => { }, TimeSpan.FromSeconds(1))
+            };
+
+            TestTimer timer = _target.CreateTimer<TestTimer>(
+                tasks,
+                new SystemTimeTracker()) as TestTimer;
+
+            AssertValidCreation(timer);
+            Assert.AreEqual(2, timer.Tasks.Count);
+        }
+
+        private void AssertValidCreation(TestTimer timer)
+        {
+            Assert.IsNotNull(timer);
+            Assert.IsTrue(timer.Initialized);
+        }
+
+        private class TestTimer : ITimer
```

Comment:
Can we break this out to its own class in the Mocks project?

Reply:
Sure, I can add the mock in this PR